contracts/amm/AmmOpenSwapService.sol

```diff
@@ -0,0 +1,819 @@
+            ammTreasury: _usdtAmmTreasury,
+            maturity: 90,
+            spreadMethodSig: "calculatePayFixed90Days((address,uint256,uint256,uint256,int256,uint256,uint256,uint256,uint256,uint256))"
+        });
+        return _openSwapPayFixed(context, totalAmount, acceptableFixedInterestRate, leverage);
+    }
+
+    function openSwapReceiveFixed28daysUsdt(
+        address onBehalfOf,
+        uint256 totalAmount,
+        uint256 acceptableFixedInterestRate,
+        uint256 leverage
+    ) external override returns (uint256) {
+        Context memory context = Context({
+            onBehalfOf: onBehalfOf,
+            asset: _usdt,
+            ammStorage: _usdtAmmStorage,
+            ammTreasury: _usdtAmmTreasury,
+            maturity: 28,
+            spreadMethodSig: "calculateReceiveFixed28Days((address,uint256,uint256,uint256,int256,uint256,uint256,uint256,uint256,uint256))"
+        });
+
+        return _openSwapReceiveFixed(context, totalAmount, acceptableFixedInterestRate, leverage);
+    }
+
+    function openSwapReceiveFixed60daysUsdt(
+        address onBehalfOf,
+        uint256 totalAmount,
+        uint256 acceptableFixedInterestRate,
+        uint256 leverage
+    ) external override returns (uint256) {
+        require(onBehalfOf != address(0), "AmmOpenSwapService: onBehalfOf is zero address");
+        Context memory context = Context({
+            onBehalfOf: onBehalfOf,
+            asset: _usdt,
+            ammStorage: _usdtAmmStorage,
+            ammTreasury: _usdtAmmTreasury,
+            maturity: 60,
+            spreadMethodSig: "calculateReceiveFixed60Days((address,uint256,uint256,uint256,int256,uint256,uint256,uint256,uint256,uint256))"
+        });
+        return _openSwapReceiveFixed(context, totalAmount, acceptableFixedInterestRate, leverage);
+    }
+
+    function openSwapReceiveFixed90daysUsdt(
+        address onBehalfOf,
+        uint256 totalAmount,
+        uint256 acceptableFixedInterestRate,
+        uint256 leverage
+    ) external override returns (uint256) {
+        Context memory context = Context({
+            onBehalfOf: onBehalfOf,
+            asset: _usdt,
+            ammStorage: _usdtAmmStorage,
+            ammTreasury: _usdtAmmTreasury,
+            maturity: 90,
+            spreadMethodSig: "calculateReceiveFixed90Days((address,uint256,uint256,uint256,int256,uint256,uint256,uint256,uint256,uint256))"
+        });
+        return _openSwapReceiveFixed(context, totalAmount, acceptableFixedInterestRate, leverage);
+    }
+
+    function openSwapPayFixed28daysUsdc(
+        address onBehalfOf,
+        uint256 totalAmount,
+        uint256 acceptableFixedInterestRate,
+        uint256 leverage
+    ) external override returns (uint256) {
+        Context memory context = Context({
+            onBehalfOf: onBehalfOf,
+            asset: _usdc,
+            ammStorage: _usdcAmmStorage,
+            ammTreasury: _usdcAmmTreasury,
+            maturity: 28,
```

Comment:
as we agreed risk management oracle doesn't match swap duration by number of days,
` @param duration swap duration, 0 = 28 days, 1 = 60 days, 2 = 90 days`